File: QytetetRuby/lib/sorpresa.rb
#Elena Cantero Molina
#encoding: utf-8
#trig.rb

module ModeloQytetet
  class Sorpresa
    attr_reader :texto, :valor, :tipo
    
    def initialize(texto, valor, tipo)
      @texto = texto
      @valor = valor
      @tipo = tipo
    end
    
    def to_s
      "Texto: #{@texto} \n Valor: #{@valor} \n Tipo: #{@tipo}"
    end
  end
end

File: QytetetJava/src/modeloqytetet/Sorpresa.java
/**
 * @author ELena Cantero Molina
 */
package modeloqytetet;

public class Sorpresa {
    private String texto;   //Describe la sorpresa
    private TipoSorpresa tipo;  //Indica el tipo de Sorpresa
    private int valor;  //valor de la sorpresa
    
    public Sorpresa(String texto, int valor, TipoSorpresa tipo){
        this.texto = texto;
        this.valor = valor;
        this.tipo = tipo;
    }
    
    //Consultor Texto de la Sorpresa
    String GetTexto(){ return texto; }
    //Consultor del valor de la sorpresa
    int GetValor(){ return valor; }
    //Consultor del tipo de sorpresa
    public TipoSorpresa GetTipoSorpresa(){ return tipo; }
    
    @Override
    public String toString(){   //Sobrecarga metodo toString
        return "Sorpresa -> " + texto + "\nvalor -> " + Integer.toString(valor) + " \ntipo -> " + tipo + "\n";
    }
}
